benchmarks/ruby-lsp/benchmark.rb

```diff
@@ -5,31 +5,28 @@
 Dir.chdir(__dir__)
 use_gemfile
 
+# This benchmark checks Ruby LSP's automatic requests. These are requests sent automatically when the user types and are
+# therefore performance critical to the LSP's responsiveness.
+
 require "ruby_lsp/internal"
 
 file_path = File.expand_path("fixture.rb", __dir__)
-file_uri = "file://#{file_path}"
-content = File.read(file_path)
+uri = URI::Generic.from_path(path: file_path)
+
+store = RubyLsp::Store.new
+store.set(uri: uri, source: File.read(file_path), version: 1)
+message_queue = Thread::Queue.new
+
+executor = RubyLsp::Executor.new(store, message_queue)
 
-rc_last = nil
-hl_last = nil
+# The Ruby LSP collects responses for all automatic requests in one go, so this is actually not only running semantic
+# highlighting, but also folding range, document symbol, code lens, etc.
 
-# These benchmarks are representative of the three main operations executed by the Ruby LSP server
 run_benchmark(200) do
-  # File parsing
-  document = RubyLsp::Document.new(content)
-
-  # Running RuboCop related requests
-  rc = RubyLsp::Requests::Diagnostics.new(file_uri, document).run
-  rc_last = rc.size
-
-  # Running SyntaxTree visitor requests
-  hl = RubyLsp::Requests::SemanticHighlighting.new(
-    document,
-    encoder: RubyLsp::Requests::Support::SemanticTokenEncoder.new,
-  ).run
-  hl_last = hl.data.size
+  executor.execute({
```

Comment:
Is there any way to check what it's returning? We had those problems where it was just not doing what was asked, and it's really good to check at least one iteration's result, if possible.

Reply:
That's fair, thanks for the context. I'll add a step that asserts the response is what we expect in all benchmarks. Indeed, it's much better to ensure the results are what we expect than getting incorrectly inflated results.